### PR TITLE
[JN-620] Parse form names into language texts

### PR DESCRIPTION
**Pull request by MatthewBemis**
#### DESCRIPTION

This adds i18n support for form names by parsing a form whenever it's saved and saving the title translations into the LanguageText table. There's also some new fallback/default functionality that ensures we're backwards compatible with existing studies.

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

* To test backwards compatibility: using the *development* branch, populate OurHealth
* Check out this branch and restart the admin and participant APIs
* Edit: make sure you rebuild ui-core
* Log in as an OurHealth participant, make sure that all of the tasks on the dashboard still have names and aren't showing up as substitution keys
* Repopulate HeartDemo
* Log in as a HeartDemo participant and cycle through the languages. Confirm that each task has a translated name in each language
* In the admin UI, make some changes to the title of The Basics survey
* Refresh the participant dashboard and confirm that you still see the original title translations for the tasks 
* Register as a new participant and confirm that you see the new translations for the latest version of the survey

### CODE REVIEW

File: core/src/main/java/bio/terra/pearl/core/service/survey/SurveyService.java [hunk 1]
```diff
@@ -74,6 +71,16 @@ public Survey create(Survey survey) {
...
             savedSurvey.getAnswerMappings().add(savedMapping);
         }
+
+        // parse the survey content to get the titles and create the language texts
+        Map<String, String> parsedTitles = SurveyParseUtils.parseSurveyTitle(savedSurvey.getContent(), savedSurvey.getName());
+        List<LanguageText> texts = SurveyParseUtils.titlesToLanguageTexts(
+                SurveyParseUtils.formToLanguageTextKey(savedSurvey.getStableId(),savedSurvey.getVersion()),
+                savedSurvey.getPortalId(),
+                parsedTitles);
+        languageTextDao.bulkCreate(texts);
```
Comment (connorlbark) [inline core/src/main/java/bio/terra/pearl/core/service/survey/SurveyService.java:85]: Why not do this parsing in the frontend? It seems indirect to save the content in a different format for every CRUD operation when the frontend (presumably) can just parse the survey content directly and grab the names. But I'm probably missing something that makes that more difficult / this a lot easier.

**Review by connorlbark — COMMENTED** [review]
🤔  Not sure if I did something wrong but I am getting substition keys after repopulating from dev then restarting admin api on the new branch.

image

Other than that, after repopulating it worked great for ourhealth and demo.

**Review by connorlbark — APPROVED** [review]
Just needed to rebuild core and now it works! looks good 👍

**Review by devonbush — APPROVED** [review]
nice, appreciate all the detail in the title parsing

**Pull request merged**